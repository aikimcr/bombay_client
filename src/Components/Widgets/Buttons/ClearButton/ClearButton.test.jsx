import React from 'react';
import { render, screen } from '@testing-library/react';

import { ClearButton } from '.';

jest.useFakeTimers();

it('should show a simple labeled input', async () => {
  const clickHandler = jest.fn();

  const { asFragment } = render(<ClearButton onClick={clickHandler} />);

  expect(asFragment).toMatchSnapshot();
});

it('should call the click handler', async () => {
  const clickHandler = jest.fn();

  render(<ClearButton onClick={clickHandler} />);

  expect(clickHandler).not.toHaveBeenCalled();

  const button = screen.getByTestId('clear-button');
  button.click();

  expect(clickHandler).toHaveBeenCalledTimes(1);
});
