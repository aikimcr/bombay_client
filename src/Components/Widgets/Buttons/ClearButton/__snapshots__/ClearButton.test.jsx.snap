// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`should show a simple labeled input 1`] = `[Function]`;
